From 69570f00548a4e7958877799b80c08d6e7474ab3 Mon Sep 17 00:00:00 2001
From: zijunzhao <zijunzhao@google.com>
Date: Thu, 22 Feb 2024 00:02:02 +0000
Subject: [PATCH] Revert: [PATCH] [BPF] Enable IAS in backend

Revert diff of commit c2d27c89593e92f8e557f27ec02f083760eeab2d:
  From: Brad Smith <brad@comstyle.com>
  Date: Sun, 5 Jun 2022 23:28:53 -0400
  Subject: [PATCH] [BPF] Enable IAS in backend
  Reviewed By: MaskRay
  Differential Revision: https://reviews.llvm.org/D123845

Change-Id: I9571d7c71e44f21417e208f095da264a69d59b6c
---
 llvm/lib/Target/BPF/MCTargetDesc/BPFMCAsmInfo.h | 2 ++
 llvm/test/CodeGen/BPF/inline_asm.ll             | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/llvm/lib/Target/BPF/MCTargetDesc/BPFMCAsmInfo.h b/llvm/lib/Target/BPF/MCTargetDesc/BPFMCAsmInfo.h
index 7b2168458c93..4243550face9 100644
--- a/llvm/lib/Target/BPF/MCTargetDesc/BPFMCAsmInfo.h
+++ b/llvm/lib/Target/BPF/MCTargetDesc/BPFMCAsmInfo.h
@@ -41,6 +41,8 @@ public:
     // section will be parsable, but with odd offsets and
     // line numbers, etc.
     CodePointerSize = 8;
+
+    UseIntegratedAssembler = false;
   }
 
   void setDwarfUsesRelocationsAcrossSections(bool enable) {
diff --git a/llvm/test/CodeGen/BPF/inline_asm.ll b/llvm/test/CodeGen/BPF/inline_asm.ll
index cdbeb4f003c9..2347416f2a34 100644
--- a/llvm/test/CodeGen/BPF/inline_asm.ll
+++ b/llvm/test/CodeGen/BPF/inline_asm.ll
@@ -34,10 +34,10 @@ entry:
   %1 = tail call i32 asm sideeffect "$0 = $1 ll", "=r,i"(i64 333333333333) #2
 ; CHECK: r1 = 333333333333 ll
   %2 = call i32 asm sideeffect "$0 = *(u16 *) $1", "=r,*m"(ptr elementtype(i32) nonnull %a) #2
-; CHECK: r1 = *(u16 *)(r10 - 4)
+; CHECK: r1 = *(u16 *) (r10 - 4)
   %3 = call i32 asm sideeffect "$0 = *(u32 *) $1", "=r,*m"(ptr elementtype(i32) getelementptr inbounds ([2 x i32], ptr @g, i64 0, i64 1)) #2
 ; CHECK: r1 = g ll
-; CHECK: r0 = *(u32 *)(r1 + 4)
+; CHECK: r0 = *(u32 *) (r1 + 4)
   call void @llvm.lifetime.end.p0(i64 4, ptr nonnull %a) #2
   ret i32 %3
 }
-- 
2.44.0.rc0.258.g7320e95886-goog

